File: week-1/75-SortColors.py
def sort_colors(nums):
    count_arr = [0] * 301
    for i in range(len(nums)):
        count_arr[nums[i]] += 1
    cursor = 0
    for k in range(len(count_arr)):
        for i in range(count_arr[k]):

            nums[i + cursor] = k
        cursor += count_arr[k]

File: 75-sort-colors/75-sort-colors.c
void swap(int * nums, int i, int j){
    int temp = nums[i];
    nums[i] = nums[j];
    nums[j] = temp;
}

void sortColors(int* nums, int numsSize){
    int smaller = 0, greater = numsSize - 1, white = 1;
    
    int i = 0;
    while(i <= greater && i < numsSize){
        if(nums[i] < white){
            swap(nums, i, smaller);
            i++;
            smaller++;
        }else if(nums[i] > white){
            swap(nums, i, greater);
            greater--;
        }else{
            i++;
        }
        
    }
}